kobo/apps/stripe/views.py

```diff
@@ -1,12 +1,137 @@
+import stripe
+
 from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from djstripe.models import Customer, Product, Subscription, Price
+from djstripe.settings import djstripe_settings
+from organizations.utils import create_organization
+from rest_framework import mixins, status, viewsets
+
 from django.db.models import Prefetch
 from djstripe.models import Plan, Product, Subscription
 from rest_framework import mixins, renderers, viewsets
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from kobo.apps.stripe.serializers import (
+    SubscriptionSerializer,
+    CheckoutLinkSerializer,
+    CustomerPortalSerializer,
+    ProductSerializer,
+)
+
+from kobo.apps.organizations.models import (
+    OrganizationUser,
+    OrganizationOwner,
+    Organization
+)
+
+class CheckoutLinkView(
+    APIView
+):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CheckoutLinkSerializer
+
+    @staticmethod
+    def generate_payment_link(price_id, user, organization_uid):
+        if organization_uid:
+            # Look up the specific organization provided
+            try:
+                organization_user = OrganizationUser.objects.get(user=user)
+                organization = Organization.objects.get(
+                    uid=organization_uid
+                )
+            # If we can't find the organization or organization user, bail
+            except ObjectDoesNotExist:
+                return Response(
+                    {'status': r'Organization does not exist'}, status=status.HTTP_403_FORBIDDEN
+                )
+        else:
+            # find the organization the user belongs to, otherwise make a new one
+            organization_user, _ = OrganizationUser.objects.get_or_create(user=user)
+            organization, _ = Organization.objects.get_or_create(users=user)
+        # Make sure the organization owner is the same as the logged-in user
+        if organization.owner.id != organization_user.id:
```

Comment:
Add this to an above query instead of making this check here. I'm pretty sure the organization.owner relation is causing yet another query. 1 query by itself isn't worth worrying about but there's a lot here. I think the code would be more readable if it was more at the database level too. 

When org uid is set, make it something like `Organization.objects.get(owner_user=user, uid=organization_uid)`
When it isn't set, the create_organization will already guarantee ownership.